src/openrct2/object/Object.h

```diff
@@ -137,60 +152,30 @@ enum class ObjectGeneration : uint8_t
 struct ObjectEntryDescriptor
 {
     ObjectGeneration Generation;
-    union
-    {
-        rct_object_entry Entry; // For DAT objects
-        char Identifier[64];    // For JSON objects
-    };
-
-    ObjectEntryDescriptor() = default;
+    std::string Identifier; // For JSON objects
+    rct_object_entry Entry; // For DAT objects
 
-    ObjectEntryDescriptor(const ObjectEntryDescriptor& entry)
+    ObjectEntryDescriptor()
+        : Generation(ObjectGeneration::JSON)
+        , Identifier()
+        , Entry()
     {
-        *this = entry;
     }
 
     explicit ObjectEntryDescriptor(const rct_object_entry& newEntry)
     {
         Generation = ObjectGeneration::DAT;
         Entry = newEntry;
+        Identifier = std::string(newEntry.GetName());
```

Comment:
Not entirely sure about this. I'd prefer having `Identifier` empty if there isn't one. (Also see the conversation in #plug-in with Bassie.)